apps/bfd-data-fda/Jenkinsfile

```diff
@@ -0,0 +1,138 @@
+        booleanParam(name: 'verbose_mvn_logging', description: 'When true, `mvn` will produce verbose logs.', defaultValue: false)
+    }
+    stages {
+        stage('Setup') {
+            steps {
+                script {
+                    container('bfd-cbc-build') {
+                        // Address limitations resulting from CVE-2022-24767
+                        sh 'git config --global --add safe.directory "$WORKSPACE"'
+
+                        // inject code artifact credentials into maven ~/.m2/settings.xml
+                        // Setup AWS Credentials
+                        awsAuth.assumeRole()
+
+                        withCredentials([string(credentialsId: 'bfd-aws-account-id', variable: 'ACCOUNT_ID')]) {
+                            codeArtifactAuthToken = sh(
+                            returnStdout: true,
+                            script: '''
+aws codeartifact get-authorization-token --domain bfd-mgmt \
+ --domain-owner "$ACCOUNT_ID" \
+ --output text --query authorizationToken
+'''
+                            ).trim()
+                            codeArtifactEndpoint = sh(
+                                returnStdout: true,
+                                script: '''
+aws codeartifact get-repository-endpoint \
+--domain bfd-mgmt --repository bfd-mgmt \
+--format maven --output text
+'''
+                            )
+                            env.CODEARTIFACT_AUTH_TOKEN = codeArtifactAuthToken
+                            env.CODEARTIFACT_ENDPOINT = codeArtifactEndpoint
+                        }
+                        sh '''
+cat <<EOF > ~/.m2/settings.xml
+<settings xmlns="http://maven.apache.org/settings/1.0.0" xmlns:xsi="http://www.w3.org/2001/xmlschema-instance"
+xsi:schemalocation="http://maven.apache.org/settings/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
+<servers>
+    <server>
+    <username>aws</username>
+    <password>${CODEARTIFACT_AUTH_TOKEN}</password>
+    <id>bfd-mgmt-bfd-mgmt</id>
+    </server>
+</servers>
+</settings>
+EOF
+'''
+                    }
+                }
+            }
+        }
+        stage('Maven Build') {
+            steps {
+                script {
+                    container('bfd-cbc-build') {
+                    // Calculate FDA DrugCode Sha256
+                        fdaDrugCodeSha256 = sh(
+                            returnStdout: true,
+                            script: '''
+curl -fsSL https://www.accessdata.fda.gov/cder/ndctext.zip | sha256sum | awk '{ print $1 }'
+'''
+                        )
+                        awsAuth.assumeRole()
+                        ssmLookupSha256 = sh(
+                            returnStdout: true,
+                            script: '''
+if parameter_lookup="$(aws ssm get-parameter --name /bfd/mgmt/versioning/nonsensitive/bfd-data-fda \
+--region us-east-1 --output text --query Parameter.Value 2>/dev/null)"; then
+   echo "$parameter_lookup"
+fi
+'''
+                        )
+
```

Comment:
A single-line comment explaining this conditional might be helpful, e.g. "Only build and deploy the artifact when changes are detected based on the sha256 hash" or similar.

Reply:
done